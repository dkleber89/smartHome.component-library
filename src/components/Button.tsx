import React, { ReactElement } from 'react';
import styled from 'styled-components';

export interface IButtonProps {
  /** Component className */
  className?: string;
}

const StyledButton = styled.div``;

StyledButton.displayName = 'StyledButton';

const Button = ({ className }: IButtonProps): ReactElement => {
  return <StyledButton className={className} />;
};

export { Button };
