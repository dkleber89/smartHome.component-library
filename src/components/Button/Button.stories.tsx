import React, { ReactElement } from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';

import { Button } from '@material-ui/core';

export default {
  title: 'Button',
  component: Button,
  decorators: [withKnobs],
  parameters: {
    info: `
    ### Additional Markdown Information about this Component
    `,
  },
};

export const Default = (): ReactElement => (
  <Button color="primary" variant="contained">
    Primary
  </Button>
);

export const DefaultSecondary = (): ReactElement => (
  <Button color="secondary" variant="contained">
    Secondary
  </Button>
);

export const Default1 = (): ReactElement => (
  <Button color="primary" variant="outlined">
    Primary
  </Button>
);

export const DefaultSecondary1 = (): ReactElement => (
  <Button color="secondary" variant="outlined">
    Secondary
  </Button>
);
