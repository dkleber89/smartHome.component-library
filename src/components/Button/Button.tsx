import React, { ReactElement } from 'react';
import styled from 'styled-components';
import { Button as MuiButton } from '@material-ui/core';

export interface IButtonProps {
  /** Component className */
  className?: string;
}

const StyledButton = styled(MuiButton)<IButtonProps>`
  background-color: ${props => props.theme.palette.primary.light};
`;

StyledButton.displayName = 'StyledButton';

const Button = ({ className }: IButtonProps): ReactElement => {
  return <StyledButton className={className}>Fisch</StyledButton>;
};

export { Button };
