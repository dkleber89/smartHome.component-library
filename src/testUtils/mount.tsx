import { createMount } from '@material-ui/core/test-utils';
import React, { ReactElement } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { StylesProvider, CssBaseline, ThemeProvider as MuiThemeProvider, Theme } from '@material-ui/core';
import lightTheme from '../themes/lightTheme';

function mountWithTheme<compProps>(tree: ReactElement<compProps>, theme: Theme = lightTheme): ReactWrapper<compProps> {
  const mount = createMount();

  const ThemeProviderWrapper = (children: ReactElement) => {
    return (
      <StylesProvider injectFirst>
        <StyledThemeProvider theme={theme}>
          <MuiThemeProvider theme={theme}>
            <CssBaseline />
            {children}
          </MuiThemeProvider>
        </StyledThemeProvider>
      </StylesProvider>
    );
  };

  return mount(tree, { wrappingComponent: ThemeProviderWrapper });
}

export default mountWithTheme;
