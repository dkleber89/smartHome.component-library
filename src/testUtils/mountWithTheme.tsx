import React, { ReactElement, ReactChild } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { StylesProvider, ThemeProvider as MuiThemeProvider, Theme } from '@material-ui/core';
import { ReactWrapper, mount } from 'enzyme';
import { ITheme, lightTheme } from '../themes';

interface IThemeProviderWrapperProps {
  children: ReactChild;
}

function mountWithTheme<compProps>(tree: ReactElement<compProps>, theme: Theme & ITheme = lightTheme): ReactWrapper<compProps> {
  const ThemeProviderWrapper = ({ children }: IThemeProviderWrapperProps): ReactElement => {
    return (
      <StylesProvider injectFirst>
        <StyledThemeProvider theme={theme}>
          <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
        </StyledThemeProvider>
      </StylesProvider>
    );
  };

  return mount<compProps>(tree, { wrappingComponent: ThemeProviderWrapper });
}

export default mountWithTheme;
