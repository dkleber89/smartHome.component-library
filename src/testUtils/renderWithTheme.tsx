import React, { ReactElement } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider, Theme } from '@material-ui/core';
import { render } from 'enzyme';
import { ITheme, lightTheme } from '../themes';

interface IThemeProviderWrapperProps {
  children: ReactElement;
}

function renderWithTheme<compProps>(tree: ReactElement<compProps>, theme: Theme & ITheme = lightTheme): Cheerio {
  React.useLayoutEffect = React.useEffect;

  const ThemeProviderWrapper = ({ children }: IThemeProviderWrapperProps): ReactElement => {
    return (
      <StylesProvider injectFirst>
        <StyledThemeProvider theme={theme}>
          <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
        </StyledThemeProvider>
      </StylesProvider>
    );
  };

  return render<compProps, never>(tree, { wrappingComponent: ThemeProviderWrapper });
}

export default renderWithTheme;
