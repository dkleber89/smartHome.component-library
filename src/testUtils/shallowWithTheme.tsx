import React, { ReactElement } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { StylesProvider, ThemeProvider as MuiThemeProvider, Theme } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import { ITheme, lightTheme } from '../themes';

interface IThemeProviderWrapperProps {
  children: ReactElement;
}

function shallowWithTheme<compProps>(tree: ReactElement<compProps>, theme: Theme & ITheme = lightTheme): ShallowWrapper<compProps> {
  const ThemeProviderWrapper = ({ children }: IThemeProviderWrapperProps): ReactElement => {
    return (
      <StylesProvider injectFirst>
        <StyledThemeProvider theme={theme}>
          <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
        </StyledThemeProvider>
      </StylesProvider>
    );
  };

  return shallow<compProps>(tree, { wrappingComponent: ThemeProviderWrapper });
}

export default shallowWithTheme;
