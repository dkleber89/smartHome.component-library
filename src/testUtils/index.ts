import mountWithTheme from './mountWithTheme';
import renderWithTheme from './renderWithTheme';
import shallowWithTheme from './shallowWithTheme';

export { mountWithTheme, renderWithTheme, shallowWithTheme };
