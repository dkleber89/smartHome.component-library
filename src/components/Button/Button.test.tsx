/**
 * @jest-environment node
 */
import React from 'react';
import { shallowWithTheme, mountWithTheme, renderWithTheme } from '../../testUtils';

import { Button, IButtonProps } from './Button';

describe('Button Snapshottesting', () => {
  /* @jest-environment node */

  test('Test', () => {
    const wrapper = renderWithTheme<IButtonProps>(<Button />);

    expect(wrapper).toMatchSnapshot();
  });
});
