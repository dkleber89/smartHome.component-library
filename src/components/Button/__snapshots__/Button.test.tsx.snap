// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshottesting Test 1`] = `
.c0 {
  background-color: red;
}

<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text c0"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    Fisch
  </span>
</button>
`;
